### DISCUSSION TITLE
convergence problem

### DISCUSSION THREAD

**Discussion by plumbum082** [deepmodeling/deepks-kit#22] [Q&A] [answered]
Hi, I encounter convergence problems when training the water cluster model. However, different from the example, I use the energy data calculated with ccsd(t)/avdz and avqz and I do not use force labels.
The problem is that the data does not converge in every SCF calculation except for the initial iteration. I mean the convergence in log.scf for many of the data points is False.
At first, I use a large amount of data, then I reduced the size to be comparable with the example data set, but it still does not converge. For example, in ./iter.00/00.scf/log.data, the file says:

Training:
  Convergence:
    67 / 300 = 	 0.22333
  Energy:
    ME: 	 0.40749853427277183
    MAE: 	 0.40749853427277183
    MARE: 	 0.0752403900542129
Testing:
  Convergence:
    1 / 50 = 	 0.02000
  Energy:
    ME: 	 0.6089715041279362
    MAE: 	 0.6089715041279362
    MARE: 	 0.2014729698551644

I failed to figure out what is the problem and I am not sure whether I should change some parameters in args.yaml, such as certain keys in [scf_args](https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L28), or [basis](https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L18) ? 
What can be the possibilities? By the way, I have tried the 'embedding: [thermal'](https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L162) and it does not work. 
Could anyone help me? @y1xiaoc  Could you please help me?

**Comment by y1xiaoc** [comment, marked as answer]
Thanks for the question! For water system the convergence should be rather easy. Have you tried with original label included in the example? If the original label works yet the current one doesn't, then there might be something wrong with the label you use (e.g., it contains some noise or is generated from different settings). Maybe you can try to reduce the label to ccsd/avdz as a first attempt?

On the other hand, you might want to try to set the [basis](https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L18) to the one you used in calculating your label (avdz for example) or try to reduce the [initial training epochs](https://github.com/deepmodeling/deepks-kit/blob/9bbec5232425e54ed7bb8cb25201d0a8f3f50281/examples/water_cluster/args.yaml#L179) to 5000. Both might be helpful to alleviate the convergence problem.

**Reply by plumbum082** [reply]
Thanks a lot! Yes, the original label included in the example works but the current one doesn't. If I would like to change the basis to avdz for example, then in what format could I change the basis? I mean I am not sure whether I could just change 'ccpvdz' to 'avdz' or I should provide a file and write the path to it.

**Reply by y1xiaoc** [reply]
It takes the same string format as pyscf. You may write "aug-ccpvdz" for example. But I would suggest you check the label first. How did you generate them?

**Reply by plumbum082** [reply]
Okay, thank you very much! l'll have a try. The labels were generated using molpro with avdz and avqz basis set. I checked the energy, they are a little more negative than the labels in the example, for example for dimers about 0.3 hartree more negative.

**Reply by y1xiaoc** [reply]
Just want to make sure: you mentioned avdz and avqz labels. Did you train on both of them? Or did you train separate models for the two? It is not possible for a single model to learn labels from different basis sets.

**Reply by plumbum082** [reply]
Sorry about that, I mean total energy of water dimers was calculated with avqz, others like trimers, etc were calculated with avdz. For example, I use the dimers and trimers as the training set, pentamers as the validation set.

**Reply by y1xiaoc** [reply]
Then this should be the cause of the issue. You will need to make sure the all the labels in the training set are generated from same settings (methods, basis sets, etc.) otherwise there is a significant discontinuity in the data and the model is not able to converge.

**Reply by plumbum082** [reply]
Okay, I understand. Thanks a lot for your help!
